discovery_engine_core/core/src/storage.rs

```diff
@@ -151,6 +155,17 @@ pub(crate) trait SourcePreferenceScope {
     async fn fetch_excluded(&self) -> Result<HashSet<String>, Error>;
 }
 
+#[async_trait]
+pub(crate) trait SourceReactionScope {
+    async fn fetch_source_reaction(&self, source: &str) -> Result<Option<bool>, Error>;
```

Comment:
❔ why not fetch whole source with `weight` and `liked`? we don't plan to use it anywhere?

Reply:
there will probably be a "fetch all" or "fetch best source" in a later PR so the `weight` in particular will be used then!